### PR TITLE
[14.0] [IMP] `base_product_mass_addition`: avoid writing on product's metadata + add tests

### PR DESCRIPTION
**Pull request by ivantodorovich**
Supress `LOG_ACCESS_COLUMNS` writes if we're only writing on quick magic fields, as they could lead to concurrency issues.

Moreover, from a functional perspective, these magic fields aren't really modifying the product's data so it doesn't make sense to update its metadata.

We achieve it by reverting the changes made by ``write`` [^1], before [^2] reaching any explicit flush [^3] or inverse computation [^4].

[^1]: https://github.com/odoo/odoo/blob/f74434c6f/odoo/models.py#L3652-L3663
[^2]: https://github.com/odoo/odoo/blob/f74434c6f/odoo/models.py#L3686
[^3]: https://github.com/odoo/odoo/blob/f74434c6f/odoo/models.py#L3689
[^4]: https://github.com/odoo/odoo/blob/f74434c6f/odoo/models.py#L3703

Basically, if all we're modifying are quick magic fields, and we don't have any other column to flush besides the `LOG_ACCESS_COLUMNS`, clear it.

### CODE REVIEW

File: base_product_mass_addition/models/product_product.py [hunk 1]
```diff
@@ -37,6 +38,34 @@ def _inverse_set_process_qty(self):
...
+        # reaching any explicit flush [^2] or inverse computation [^3].
+        #
+        # [^1]: https://github.com/odoo/odoo/blob/f74434c6f/odoo/models.py#L3652-L3663
+        # [^2]: https://github.com/odoo/odoo/blob/f74434c6f/odoo/models.py#L3686
+        # [^3]: https://github.com/odoo/odoo/blob/f74434c6f/odoo/models.py#L3689
+        # [^4]: https://github.com/odoo/odoo/blob/f74434c6f/odoo/models.py#L3703
+        #
+        # Basically, if all we're modifying are quick magic fields, and we don't have
+        # any other column to flush besides the LOG_ACCESS_COLUMNS, clear it.
+        quick_fnames = {"qty_to_process", "quick_uom_id"}
```
Comment (simahawk) [inline base_product_mass_addition/models/product_product.py]: pls use a `tuple` in these cases

File: base_product_mass_addition/tests/models/__init__.py [hunk 2]
```diff
@@ -0,0 +1 @@
+from . import order
```
Comment (simahawk) [inline base_product_mass_addition/tests/models/__init__.py]: This must be imported ONLY in the test class setup

**Review by simahawk — CHANGES_REQUESTED** [review]

File: base_product_mass_addition/models/product_product.py (hunk 1 above)
Comment (ivantodorovich) [reply base_product_mass_addition/models/product_product.py]: done 👍🏻

File: base_product_mass_addition/tests/models/__init__.py (hunk 2 above)
Comment (ivantodorovich) [reply base_product_mass_addition/tests/models/__init__.py]: done 👍🏻

**Review by gurneyalex — COMMENTED** [review]
The code looks ok to me. I'm just wondering if this is the right place to put this log silencing code? Should this not live in either a customer specific module or in an entirely separate specific module?

**Comment by ivantodorovich** [comment]
> The code looks ok to me. I'm just wondering if this is the right place to put this log silencing code? Should this not live in either a customer specific module or in an entirely separate specific module?

IMO this is the right place because these "magic fields" aren't really meant to write product data. It's a rather hacky way to achieve what we really want: being able to create/update `sale|purchase.order.line` from the product tree view.

**Review by damdam-s — APPROVED** [review]
thanks

**Comment by legalsylvain** [comment]
@simahawk, could you update your review ? your remarks looks answered.

thanks !

**Review by simahawk — APPROVED** [review]
